roxie/roxiemem/roxiemem.cpp

```diff
@@ -1955,19 +1983,22 @@ class HugeHeaplet : public Heaplet
 
     virtual void noteReleased(const void *ptr)
     {
+        //NORE: See comment on FixedSizeHeaplet::noteReleased() regarding the memory order operands
+
         //If rowCount == 1 then this must be the last reference - avoid a locked operation.
         //rowCount is not used once the heaplet is known to be freed, so no need to ensure rowCount=0
-        if (atomic_read(&rowCount) == 1 || atomic_dec_and_test(&rowCount))
+        if (rowCount.load(std::memory_order_relaxed) == 1 || rowCount.fetch_sub(1, std::memory_order_release) == 1)
```

Comment:
I think it is correct.  dec_and_test decrements the value, and then returns it.  fetch_sub gets the value and then subtracts it.  The c++ standard has no sub_fetch which would be the closest equivalent.

Previous definition:
static __inline__ bool atomic_inc_and_test(atomic_t *v)
{
    return (__sync_add_and_fetch(&v->counter,1)==0);
}